discord/ext/tasks/__init__.py

```diff
@@ -562,7 +562,7 @@ def before_loop(self, coro: FT) -> FT:
             raise TypeError(f'Expected coroutine function, received {coro.__class__.__name__!r}.')
 
         self._before_loop = coro
-        return coro
+        return coro  # type: ignore
```

Comment:
Can we add an explanation comment for this and similar ones?

I feel like this one's a pyright thing since this is just an identity function...

Reply:
I did a bit of digging here and `inspect.iscoroutinefunction()` changes the type to `(...) -> types.CoroutineType[Any, Any, Any]` which is not compatible with `FT` (bound to an `Awaitable[]` return). There are two ways to fix this depending on what the goal is:

* Change `LF`, `FT`, and `ET` to be bound to a callable that returns `Coroutine[Any, Any, Any]`. If we want the code to stay the same as it is right now, this is the correct solution.
* Change the task code to check if the return value of the functions is awaitable rather than checking if the function itself is a coroutine function. If we want `Loop` to be able to use functions returning awaitables, this is the correct solution.

Let me know what you'd like me to do.

Reply:
The first one is the correct solution. These are decorators so passing in sync functions that return awaitables is not really the intended use case since it'd be hard to check for this in a clean manner (some awaitables can only be awaited once or may return a type that causes some kind of limitation in terms of use).